ios/RNSentryDependencyContainer.m

```diff
@@ -0,0 +1,35 @@
+#import "RNSentryDependencyContainer.h"
+#import <Sentry/SentryDependencyContainer.h>
+
+@implementation RNSentryDependencyContainer
+
+static RNSentryDependencyContainer *instance;
+static NSObject *sentryDependencyContainerLock;
+
++ (void)initialize
+{
+  if (self == [RNSentryDependencyContainer class]) {
+    sentryDependencyContainerLock = [[NSObject alloc] init];
+    instance = [[RNSentryDependencyContainer alloc] init];
+  }
+}
+
++ (instancetype)sharedInstance
+{
+  return instance;
+}
+
+- (instancetype)init
+{
+    return self;
+}
```

Comment:
l: A good practice for singletons in Objc would be like the following. 
Your code is good, so no need to change it if you dont want to, just a objc tip.

```suggestion
@implementation RNSentryDependencyContainer {
     NSObject *sentryDependencyContainerLock;
 }

+ (instancetype)sharedInstance
{
    static RNSentryDependencyContainer *instance = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{ instance = [[self alloc] init]; });
    return instance;
}

- (instancetype)init
{
    if (self = [super init]) {
        sentryDependencyContainerLock = [[NSObject alloc] init];
    }
    return self;
}
```